src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,160 @@
+  private double maxEAngle = Units.degreesToRadians(89);
+  private double baseArmMass = Units.lbsToKilograms(15); //15
+  private double elbowArmMass = Units.lbsToKilograms(15);
+  private final double baseArmInertia = SingleJointedArmSim.estimateMOI(baseArmLength, baseArmMass);
+  private final double elbowArmInertia = SingleJointedArmSim.estimateMOI(elbowArmLength, elbowArmMass);
+  private double simBCurrentAngle;
+  private double simECurrentAngle;
+
+  private final SingleJointedArmSim baseArmSim = new SingleJointedArmSim(DCMotor.getNEO(2), baseGearing, baseArmInertia, baseArmLength, minBAngle, maxBAngle, baseArmMass, false);
+  private final SingleJointedArmSim elbowArmSim = new SingleJointedArmSim(DCMotor.getNEO(1), elbowGearing, elbowArmInertia, elbowArmLength, minEAngle, maxEAngle, elbowArmMass, false);
+
+  public ArmSubsystem() {
+    motorBaseOne = MotorController.constructMotor(MotorConfig.ArmBaseMotor1);
+    motorBaseTwo = MotorController.constructMotor(MotorConfig.ArmBaseMotor2);
+    motorBaseTwo.follow(motorBaseOne);
+    motorElbow = MotorController.constructMotor(MotorConfig.ArmElbowMotor);
+    //TODO: Place real values for channels
+    motorBaseOneEncoder = new Encoder(0,1);
+    motorElbowEncoder = new Encoder(2,3);
+    motorBaseOneEncoder.setDistancePerPulse(1./baseGearing); //Amount of distance per rotation of the motor, basically a gear ratio measurement
+    motorElbowEncoder.setDistancePerPulse(1./elbowGearing);
+    motorBaseOneEncoderSim = new EncoderSim(motorBaseOneEncoder);
+    motorElbowEncoderSim = new EncoderSim(motorElbowEncoder);
+    basePIDController = new PIDController(BP, BI, BD);
+    elbowPIDController = new PIDController(EP, EI, ED);
+  }
+
+  @Override
+  public void periodic() {  
+    // This method will be called once per scheduler run
+  }
+
+  public void setBaseRef(double setpoint) {
+    motorBaseOne.set(basePIDController.calculate(motorBaseOneEncoder.getDistance(), setpoint));
+  }
+  public double getBaseAngle() {
+    return motorBaseOneEncoder.getDistance() * (Math.PI*2);
+  }
+
+   public void setElbowRef(double setpoint) {
+    motorElbow.set(elbowPIDController.calculate(motorElbowEncoder.getDistance(), setpoint));
+  }
+  public double getElbowAngle() {
+    return motorElbowEncoder.getDistance() * (Math.PI*2);
+  }
+
+  @Override
+  public void simulationPeriodic() {
+    //Base arm angle
+    double simulationXCoord = 100*Units.inchesToMeters(45.8);
+    double simulationYCoord = 100*Units.inchesToMeters(26.9);
+    double baseSetpoint = ArmAutoCommand.getBaseAngle(simulationXCoord, simulationYCoord)+Units.degreesToRadians(0);
+    double elbowSetpoint = ArmAutoCommand.getElbowAngle(simulationXCoord, simulationYCoord)+Units.degreesToRadians(0);
+    double basePidOut = basePIDController.calculate(motorBaseOneEncoderSim.getDistance(), baseSetpoint);
+    double elbowPidOut = elbowPIDController.calculate(motorElbowEncoderSim.getDistance(), elbowSetpoint);
```

Comment:
There's a unit mismatch here. Encoder values are in rotations, but the setpoints are in radians

Reply:
fixed by changing setDistancePerPulse to output radians